app/actions/remote/search.ts

```diff
@@ -131,26 +130,11 @@ export const searchPosts = async (serverUrl: string, params: PostSearchParams):
 };
 
 export const searchFiles = async (serverUrl: string, teamId: string, params: FileSearchParams): Promise<FileSearchRequest> => {
-    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
-
-    if (!operator) {
-        return {error: `${serverUrl} database not found`};
-    }
-
-    let client: Client;
-    try {
-        client = NetworkManager.getClient(serverUrl);
-    } catch (error) {
-        return {error};
-    }
-
-    let data;
     try {
-        data = await client.searchFiles(teamId, params.terms);
+        const client = NetworkManager.getClient(serverUrl);
+        return await client.searchFiles(teamId, params.terms);
```

Comment:
In theory, you don't need the `await` since this function returns a promise either way.

But... (and I am not 100% sure of this) you may want to await the result in case the function throws, so the try catch can intercept that.

I am not sure if this `return await` awaits here in the function (and therefore properly goes through the catch branch) or will return the promise, and skip the try/catch branch.